File: Platforms/Windows/Reign.Video.D3D9/VertexShader.cpp
#include "pch.h"
#include "VertexShader.h"
#include "Shader.h"

using namespace System::Runtime::InteropServices;
using namespace Reign::Core;

namespace Reign
{namespace Video
{namespace D3D9
{
	#pragma region Constructors
	VertexShader::VertexShader(ShaderI^ shader, string^ code, ShaderVersions shaderVersion)
	: ShaderModel(shader, code, shaderVersion, ShaderTypes::VS)
	{
		null();
		try
		{
			IDirect3DVertexShader9* shaderTEMP = 0;
			video->Device->CreateVertexShader((DWORD*)this->code->GetBufferPointer(), &shaderTEMP);
			this->shader = shaderTEMP;
		}
		catch (Exception^ ex)
		{
			delete this;
			throw ex;
		}
	}

	VertexShader::~VertexShader()
	{
		disposeChilderen();
		if (shader) shader->Release();
		null();
	}

	void VertexShader::null()
	{
		shader = 0;
	}
	#pragma endregion

	#pragma region Methods
	void VertexShader::Apply()
	{
		video->Device->SetVertexShader(shader);
	}
	#pragma endregion
}
}
}

File: Platforms/Common/Reign.Video.OpenGL/Shaders/VertexShader.cs
using System;
using Reign.Core;
using System.Collections.Generic;

namespace Reign.Video.OpenGL
{
	public class VertexShader : ShaderModel
	{
		#region Constructors
		public VertexShader(IShader shader, string code, ShaderVersions shaderVersion)
		: base(shader, code, shaderVersion, ShaderTypes.VS, ShaderFloatingPointQuality.High)
		{

		}
		
		public VertexShader(IShader shader, string code, ShaderVersions shaderVersion, ShaderFloatingPointQuality quality)
		: base(shader, code, shaderVersion, ShaderTypes.VS, quality)
		{
			
		}
		#endregion
	}
}